sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java

```diff
@@ -195,8 +197,12 @@ public void create() {
      */
     public Response<Void> createWithResponse(Map<String, String> metadata, PublicAccessType accessType,
         Duration timeout, Context context) {
-        return blobContainerClient.createWithResponse(metadata, Transforms.toBlobPublicAccessType(accessType), timeout,
-            context);
+        try {
+            return blobContainerClient.createWithResponse(metadata, Transforms.toBlobPublicAccessType(accessType),
+                timeout, context);
+        } catch (BlobStorageException ex) {
+            throw logger.logExceptionAsError(DataLakeImplUtils.transformBlobStorageException(ex));
+        }
```

Comment:
Could `DataLakeImplUtils` add a method such as the following to ensure the same code is called every place a try/catch convert is performed?

```java
public static <T> T returnOrConvertException(Supplier<T> supplier, ClientLogger logger) {
  try {
    return supplier.get();
  } catch (BlobStorageException ex) {
    throw logger.logExceptionAsError(transformBlobStorageException(ex));
  }
}
```

A calling pattern from the clients would look it this.

```java
return DataLakeImplUtils.returnOrConvertException(() -> 
  blobContainerClient.createWithResponse(metadata, Transforms.toBlobPublicAccessType(accessType), timeout, context), logger);
```

Just a thought, current implementation looks correct but changes could be cumbersome.

Reply:
This looks like it should work better. I'll update to this